Riot/Modules/Onboarding/AuthenticationCoordinator.swift

```diff
@@ -72,34 +72,38 @@ final class AuthenticationCoordinator: NSObject, AuthenticationCoordinatorProtoc
     // MARK: - Public
     
     func start() {
-        Task {
-            do {
-                let flow: AuthenticationFlow = initialScreen == .login ? .login : .register
-                try await authenticationService.startFlow(flow, for: authenticationService.state.homeserver.address)
-            } catch {
-                MXLog.error("[AuthenticationCoordinator] start: Failed to start")
-                await MainActor.run { displayError(error) }
-                return
-            }
-            
-            await MainActor.run {
-                switch initialScreen {
-                case .registration:
-                    showRegistrationScreen()
-                case .selectServerForRegistration:
-                    showServerSelectionScreen()
-                case .login:
-                    showLoginScreen()
-                }
-            }
+        Task { await startAsync() }
+    }
+    
+    /// An async version of `start`.
+    ///
+    /// Allows the caller to show an activity indicator until the authentication service is ready.
+    @MainActor func startAsync() async {
```

Comment:
I'm 100% open to a better name for this 🤔

Reply:
Is this meant to be public or private only? If public, they I'd say we now have two entry-point into the coordinator, only one of which is defined by the protocol, could we make it private?

If private, then it can just have more descriptive name such as `startAuthenticationFlow`

Reply:
Updated and switched to using another case in the `callback` 👍